lib/galaxy/webapps/galaxy/api/datatypes.py

```diff
@@ -2,14 +2,148 @@
 API operations allowing clients to determine datatype supported by Galaxy.
 """
 import logging
-
-from galaxy.datatypes.data import Data
+from typing import (
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+
+from fastapi import (
+    Depends,
+    Query,
+)
+from fastapi.routing import APIRouter
+from fastapi_utils.cbv import cbv
+
+from galaxy.app import UniverseApplication
+from galaxy.datatypes.registry import Registry
+from galaxy.managers.datatypes import (
+    DatatypeConverterList,
+    DatatypeDetails,
+    DatatypesCombinedMap,
+    DatatypesMap,
+    view_converters,
+    view_edam_data,
+    view_edam_formats,
+    view_index,
+    view_mapping,
+    view_sniffers,
+    view_types_and_mapping
+)
 from galaxy.util import asbool
 from galaxy.web import expose_api_anonymous_and_sessionless
 from galaxy.webapps.base.controller import BaseAPIController
+from . import get_app
 
 log = logging.getLogger(__name__)
 
+router = APIRouter(tags=['datatypes'])
+
+ExtensionOnlyQueryParam : Optional[bool] = Query(
+    default=True,
+    title="Extension only",
+    description="Whether to return only the datatype's extension rather than the datatype's details",
+)
+
+UploadOnlyQueryParam : Optional[bool] = Query(
+    default=True,
+    title="Upload only",
+    description="Whether to return only datatypes which can be uploaded",
+)
+
+
+def get_datatypes_registry(app: UniverseApplication = Depends(get_app)) -> Registry:
+    return app.datatypes_registry
+
+
+@cbv(router)
+class FastAPIDatatypes:
+    datatypes_registry: Registry = Depends(get_datatypes_registry)
+
+    @router.get(
+        '/api/datatypes',
+        summary="Lists all available data types",
+        response_model=Union[List[DatatypeDetails], List[str]],
+        response_description="List of data types",
+    )
+    async def index(
+        self,
+        extension_only: Optional[bool] = ExtensionOnlyQueryParam,
+        upload_only: Optional[bool] = UploadOnlyQueryParam
+    ) -> Union[List[DatatypeDetails], List[str]]:
+        """Gets the list of all available data types."""
+        return view_index(self.datatypes_registry, extension_only, upload_only)
```

Comment:
These little one line methods shared with the older style controller are perfect!